File: projects/integration/workspace/api_client.py
import requests

class APIClient:
    def get(self, url, token):
        headers = {'Authorization': f'Bearer {token}'}
        response = requests.get(url, headers=headers)
        response.raise_for_status()
        return response

    def post(self, url, token, data):
        headers = {'Authorization': f'Bearer {token}'}
        response = requests.post(url, headers=headers, json=data)
        response.raise_for_status()
        return response

File: projects/9ee52fd7-13e2-4fa0-9830-79d5543e1e32/archive/20230815_124043/workspace/apiClient.js
const axios = require('axios');
const jwt = require('jsonwebtoken');

class APIClient {
  constructor(baseURL, clientId, clientSecret) {
    this.baseURL = baseURL;
    this.clientId = clientId;
    this.clientSecret = clientSecret;
    this.token = null;
  }

  async authenticate() {
    const payload = {
      iss: this.clientId,
      aud: this.baseURL,
      exp: Math.floor(Date.now() / 1000) + 60 * 60 // 1 hour from now
    };
    this.token = jwt.sign(payload, this.clientSecret);
  }

  async request(method, path, data = null, headers = {}) {
    if (!this.token) {
      await this.authenticate();
    }

    const url = `${this.baseURL}${path}`;
    const options = {
      method,
      url,
      headers: {
        ...headers,
        Authorization: `Bearer ${this.token}`
      }
    };

    if (data) {
      options.data = data;
    }

    const response = await axios(options);
    return response.data;
  }
}

module.exports = APIClient;
